From: Matthias Klose <doko@debian.org>
Date: Tue, 12 Apr 2022 18:22:51 -0400
Subject: Ubuntu: Support separate langpack packages

Support alternative gettext tree in /usr/share/locale-langpack; if a file is
present in both trees, prefer the newer one.
Ubuntu collates gettext from packages on the DVD into language packs, to
reduce disk-space on the image.
This is Ubuntu-Specific.
Forwarded: not-needed
---
 Lib/gettext.py | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/Lib/gettext.py b/Lib/gettext.py
index 77b67ae..c869b83 100644
--- a/Lib/gettext.py
+++ b/Lib/gettext.py
@@ -567,11 +567,26 @@ def find(domain, localedir=None, languages=None, all=False):
         if lang == 'C':
             break
         mofile = os.path.join(localedir, lang, 'LC_MESSAGES', '%s.mo' % domain)
+        mofile_lp = os.path.join("/usr/share/locale-langpack", lang,
+                               'LC_MESSAGES', '%s.mo' % domain)
+
+	# first look into the standard locale dir, then into the 
+	# langpack locale dir
+
+	# standard mo file
         if os.path.exists(mofile):
             if all:
                 result.append(mofile)
             else:
                 return mofile
+        
+	# langpack mofile -> use it
+        if os.path.exists(mofile_lp): 
+            if all:
+                result.append(mofile_lp)
+            else:
+                return mofile_lp
+
     return result
 
 
